fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OneRangePartitionEvaluator.java

```diff
@@ -616,6 +630,19 @@ private List<Literal> toMultiNereidsLiterals(PartitionKey partitionKey) {
         return literals;
     }
 
+    @Override
+    public EvaluateRangeResult visitDateTrunc(DateTrunc dateTrunc, EvaluateRangeInput context) {
+        EvaluateRangeResult result = super.visitDateTrunc(dateTrunc, context);
+        if (!(result.result instanceof DateTrunc)) {
+            return result;
+        }
+        Expression dateTruncChild = dateTrunc.child(0);
+        if (partitionSlotContainsNull.containsKey(dateTruncChild)) {
+            partitionSlotContainsNull.put(dateTrunc, true);
+        }
+        return computeMonotonicFunctionRange(result);
+    }
```

Comment:
if we add a new trait class Monotonic for expression, could we here use visit Monotoic to do same things for all monotonic functions? then we only need add trait on different function

Reply:
we should use trait interface
```java
public interface Monotonic extends ExpressionTrait {
  boolean isPositive(); 
}

public class DateTrunc implement Monotonic {
    @Override
    public isPositive() {
        return true;
    }
}
```